core/packages/contracts/contracts/ScaleCodec.sol

```diff
@@ -82,27 +82,27 @@ library ScaleCodec {
         v = (v >> 8) | (v << 8);
     }
 
-    function encodeU256(uint256 input) public pure returns (bytes32) {
+    function encodeU256(uint256 input) internal pure returns (bytes32) {
```

Comment:
I've already optimized ScaleCodec in my https://github.com/snowfork/snowbridge/tree/vincent/optimize-mmr branch which I'm going to make into PR soon.